users/postgres/users.go

```diff
@@ -290,17 +366,30 @@ func toUser(dbu dbUser) (users.User, error) {
 	}, nil
 }
 
-func getUsersMetadataQuery(m users.Metadata) ([]byte, string, error) {
-	mq := ""
-	mb := []byte("{}")
-	if len(m) > 0 {
-		mq = ` AND users.metadata @> :metadata`
+func createEmailQuery(entity string, email string) (string, string, error) {
+	query := ""
+	param := ""
+
+	if email != "" {
+		// Create LIKE operator to search Users with email containing a given string
+		param = fmt.Sprintf(`%%%s%%`, email)
+		query = fmt.Sprintf("%semail LIKE :email", entity)
+		return query, param, nil
+	}
+
+	return query, param, nil
+}
 
-		b, err := json.Marshal(m)
+func createMetadataQuery(entity string, meta users.Metadata) (string, []byte, error) {
+	query := ""
+	param := []byte("{}")
+	if len(meta) > 0 {
+		query = fmt.Sprintf("%smetadata @> :metadata", entity)
```

Comment:
Should this be:
```go
query = fmt.Sprintf("%s.metadata @> :metadata", entity)
```

Reply:
Maybe we should add better tests, I suspect that it would detect this kind of issues.

Reply:
No, because if entity is empty you would search `.metadata` instead of `metadata` . There are test covering this part (I had many errors building this queries) but I can add more